cmd/lakefs/cmd/run.go

```diff
@@ -51,305 +35,102 @@ type Shutter interface {
 	Shutdown(context.Context) error
 }
 
-func newLDAPAuthenticator(cfg *config.LDAP, service auth.Service) *auth.LDAPAuthenticator {
-	const (
-		connectionTimeout = 15 * time.Second
-		requestTimeout    = 7 * time.Second
-	)
-	group := cfg.DefaultUserGroup
-	if group == "" {
-		group = auth.ViewersGroup
-	}
-	return &auth.LDAPAuthenticator{
-		AuthService:       service,
-		BindDN:            cfg.BindDN,
-		BindPassword:      cfg.BindPassword,
-		DefaultUserGroup:  group,
-		UsernameAttribute: cfg.UsernameAttribute,
-		MakeLDAPConn: func(_ context.Context) (*ldap.Conn, error) {
-			c, err := ldap.DialURL(
-				cfg.ServerEndpoint,
-				ldap.DialWithDialer(&net.Dialer{Timeout: connectionTimeout}),
-			)
+func NewRootCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "run",
+		Short: "Run lakeFS",
+		Run: func(cmd *cobra.Command, args []string) {
+			logger := logging.Default()
+			cfg := loadConfig()
+			viper.WatchConfig()
+			viper.OnConfigChange(func(in fsnotify.Event) {
+				lvl := viper.GetString(config.LoggingLevelKey)
+				logger.WithField("toLevel", lvl).Info("Changing log level")
+				logging.SetLevel(lvl)
+			})
+			ctx := cmd.Context()
+			lakeFSCmdCtx := application.NewLakeFSCmdContext(cfg, logger)
+			logger.WithField("version", version.Version).Info("lakeFS run")
+			databaseService := application.NewDatabaseService(ctx, lakeFSCmdCtx)
+			err := databaseService.ValidateSchemaIsUpToDate(ctx, lakeFSCmdCtx)
+			if err != nil && errors.Is(err, db.ErrSchemaNotCompatible) {
+				os.Exit(1)
+			}
```

Comment:
The logging is done in the function, because it was duplicated I refactored it. If I extract it I will re-duplicate it (run and import). Are you ok with that?